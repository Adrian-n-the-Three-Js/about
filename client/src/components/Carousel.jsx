import React from 'react';
// import CarouselContent from './CarouselContent.jsx';
import Photo from './Photo.jsx';
import CarouselPhotostrip from './CarouselPhotostrip.jsx';
import Arrow from './Arrow.jsx';
import styled from 'styled-components';

console.log('carousel');

class Carousel extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      currentPhotoIndex: 0,
    };
    this.previousPhoto = this.previousPhoto.bind(this);
    this.nextPhoto = this.nextPhoto.bind(this);
  }

  previousPhoto() {
    console.log('previous photo');
    const lastIndex = this.props.preview.length - 1;
    const currentPhotoIndex = this.state.currentPhotoIndex;
    const index = currentPhotoIndex !== 0 ? currentPhotoIndex - 1 : null;
    this.setState({
      currentPhotoIndex: index
    });
  }

  nextPhoto() {
    console.log('next photo');
    const lastIndex = this.props.preview.length - 1;
    const currentPhotoIndex = this.state.currentPhotoIndex;
    const index = currentPhotoIndex !== lastIndex ? currentPhotoIndex + 1 : null;
    this.setState({
      currentPhotoIndex: index
    });
  }

  render() {
    // this.props.preview.length && console.log('image', this.props.hotel[0]);
    return (
      <Wrapper>

        {(this.state.currentPhotoIndex - 1 >= 0) && (
        <Arrow
          direction="left"
          symbol="&#9664;"
          onClick={this.previousPhoto}
        />
        )}

        {/* {
          this.props.preview.length &&
          <img src={this.props.preview[this.state.currentPhotoIndex]['imageUrl']}/>
        } */}

        {(this.state.currentPhotoIndex + 1 <= this.props.preview.length - 1) && (
        <Arrow
          direction="right"
          symbol="&#9654;"
          onClick={this.nextPhoto}
        />
        )}

        <PhotoContainer>
          <Photo url={this.props.preview[this.state.currentPhotoIndex].imageUrl} />
        </PhotoContainer>

        <PhotostripWrapper>
          {
          this.props.preview.map((one, index) => (
            <CarouselPhotostrip
              // key={one._id}
              index={index}
              photo={one.imageUrl}
              caption={one.caption}
            />
          ))
          }
        </PhotostripWrapper>

      </Wrapper>
    );
  }
}

const Wrapper = styled.div`
  background-color: #f5dcdc;
  // display: block;
  // flex-direction: column;
  cursor: pointer;
  height: 330px;
  width: 330px;
  // padding-top: 50px;
  // padding-right: 80px;
  // padding-bottom: 50px;
  // padding-left: 80px;
  // margin: 0 auto;
  // overflow: hidden;
`;

const PhotoContainer = styled.div`
  height: 270px;
  width: 370px;
`;

const PhotostripWrapper = styled.div`
  display: flex;
  height: 50px;
  width: 370px;

  // display: flex;
  // flex-wrap: wrap;
  // border: 1px solid blue;
  // max-width: 50px;
  // max-height: 50px;
  // margin: 1px -1px -1px;
  // padding: 0;
  // cursor: pointer;
  // // height: 5px;
  // // width: 5px
  // overflow: hidden;
`;

const PhotoStrip = styled(PhotostripWrapper)`
  position: relataive:
  width: 100px;
  height: 100px;

  // margin: 0 auto;
  // &:after {
  //   content: '';
  //   display: block;
  //   width: 100%;
  //   height: 265px;
  //   outline: 5px solid #fc2003;
  //   position: absolute;
  //   top: 0;
  //   left: 0;
  }
`;

export default Carousel;
