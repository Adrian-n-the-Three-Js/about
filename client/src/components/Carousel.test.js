import React from 'react';
import { mount, shallow, render } from 'enzyme';
import Carousel from './Carousel.jsx';

describe ('Carousel', () => {

  // let CarouselComponent;
  // beforeEach( () => {
  //   CarouselComponent = shallow(<Carousel data={[{id: 1, user: 'name'}, {id: 2, user: test}]} />);
  // });
    // CarouselComponent = shallow(<Carousel data={
    //   [{"_id": "1",
    //     "roomAlbum": ["_id":"001","user":"George","userAvatarURL":"www.google.com","caption":"Cat","category":"Room","datePosted":"2020-05-23T15:15:51-04:00","helpfulVotes":1}],

    //     "diningAlbum":[{"_id":"001","user":"Dan Gold","userAvatarURL":"www.google.com","caption":"Friends","category":"Dining","datePosted":"2016-06-26T18:09:09-04:00","helpfulVotes":6},{"_id":"002","user":"Zakaria Zayane","userAvatarURL":"www.google.com","caption":null,"category":"Dining","datePosted":"2019-02-23T19:22:42-05:00","helpfulVotes":6}],

    //     "poolAlbum":[{"_id":"001","user":"MILKOVÍ","userAvatarURL":"www.google.com","caption":null,"category":"Pool & Beach","datePosted":"2019-05-09T23:35:28-04:00","helpfulVotes":2}],

    //     "gymAlbum":[{"_id":"001","user":"Spencer Davis","userAvatarURL":"www.google.com","caption":null,"category":"Gym","datePosted":"2019-01-28T11:19:13-05:00","helpfulVotes":5},{"_id":"5eed5e121b81ed33e65a8b9e","user":"Humphrey Muleba","userAvatarURL":"https://images.unsplash.com/profile-1533938603496-a558c9b54f56?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128","imageUrl":"https://images.unsplash.com/photo-1540497077202-7c8a3999166f?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjE0Mjk4NH0","caption":"instagram @good.citizen","category":"Gym","datePosted":"2018-10-25T15:52:58-04:00","helpfulVotes":1},{"_id":"5eed5e121b81ed33e65a8b9f","user":"Javier Santos Guzmán","userAvatarURL":"https://images.unsplash.com/profile-1580672464502-84958a7e5224image?ixlib=rb-1.2.1&q=80&fm=jpg&crop=faces&cs=tinysrgb&fit=crop&h=128&w=128","imageUrl":"https://images.unsplash.com/photo-1580261450046-d0a30080dc9b?ixlib=rb-1.2.1&q=80&fm=jpg&crop=entropy&cs=tinysrgb&w=1080&fit=max&ixid=eyJhcHBfaWQiOjE0Mjk4NH0","caption":"www.javiersantos.blog ","category":"Gym","datePosted":"2020-01-28T20:31:22-05:00","helpfulVotes":2}],"__v":0
    // ]} />);
  // });

  xit('should render correctly', () => {
    const CarouselComponent = shallow(<Carousel />);
    expect(CarouselComponent).toMatchSnapshot();
  });

  xit('should render photo preview', () => {
    const photoPreview = CarouselComponent.find('CarouselPhotostrip');
    expect(photoPreview.length).toEqual(1);
  })

  xit('should update state', () => {
    const CarouselComponent = shallow(<Carousel
      preview={
        {preview: [{_id: 1, imageUrl: 'www.google.com'}] }
      }
      currentPhotoIndex={ {test: 0} }
    />);
    expect(CarouselComponent.state('currentPhotoIndex')).toBe(0);
    // const CarouselComponent = mount(<Carousel />);
    // const startState = {currentPhotoIndex: 0}
    // const newState = Carousel.nextPhoto();
    // expect(newState.currentPhotoIndex).toEqual(1);
  });

});
