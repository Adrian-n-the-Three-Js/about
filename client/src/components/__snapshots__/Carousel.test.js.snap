// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel should render correctly 1`] = `
<div>
  <Carousel__PhotoContainer>
    <Photo />
    <AlbumPhotoCount />
  </Carousel__PhotoContainer>
  <Carousel__PhotostripWrapper>
    <CarouselPhotostrip
      className="photostrip"
      index={0}
      onClick={[Function]}
    />
  </Carousel__PhotostripWrapper>
</div>
`;
